tests/basic/misc_test.go

```diff
@@ -75,6 +148,11 @@ var _ = Describe("{VolumeDriverDown}", func() {
 
 		ValidateApplications(contexts)
 
+		var pureCleanupFunction func()
+		if Inst().V.String() == portworx.PureDriverName {
+			pureCleanupFunction = StartPureBackgroundWriteRoutines()
```

Comment:
I guess this is the same readability nitpick: the variable name "cleanupFunction" seems not quite aligned with what the function StartPureBackgroundWriteRoutines() does? The purpose of pureCleanupFunction isn't very obvious to me

Reply:
`pureCleanupFunction` is basically what stops the goroutines and checks for errors. So the "start" function returns that cleanup function so we can then stop everything, without needing a global state variable

Reply:
hmm I found it a little unintuitive cuz `pureCleanupFunction()` doesn't do any "cleanup" work.

Reply:
This question is really for my education: what are some reasons we prefer defining the procedure to stop writing as a function returned from StartPureBackgroundWriteRoutines() over using a global state variable?

Reply:
Good point on the name, I guess "cleanup" for me was that it was "cleaning up" (stopping) the routines. `stopBackgroundWrites` would be more clear, I agree.

In my opinion, I opted for this because there's multiple tests using the same logic. If it was just one test, I was going to keep it as a boolean, but since both VDDown and VDCrash will use this function, I was concerned about bugs from not resetting the state properly. This way, we can just use local variables and closure them, so that they never leave the context of the test.